### PR TITLE
add invariant document clarification

### PR DESCRIPTION
**Pull request by VoR0220**
This adds more clarification around the invariant documentation, attempts to deduplicate from the docs wherever possible, establish the proper links and clarify where tests for invariants are implemented and where they are not.

### CODE REVIEW

File: docs/userGuides/development/invariants/client/token/ERC20/ProtocolERC20-invariants.md [hunk 1]
```diff
@@ -1,7 +1,36 @@
...
+- TransferFroms for more than available balance should not be allowed
+- Zero amount transfers should not break accounting
+- Zero amount transferFroms should not break accounting
+- Transfers should update accounting correctly
+- TransferFroms should update accounting correctly
+- Approve should set correct allowances
+- Allowances should be updated correctly when approve is called twice.
+- TransferFrom should decrease allowance
+
+## [Burnable ERC20 Invariants](../../../../../../../test/client/token/ERC20/invariant/ApplicationERCMintBurn.t.i.sol)
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/token/ERC20/ProtocolERC20-invariants.md]: I think this link might be off. Should be ApplicationERC20MintBurn

File: docs/userGuides/development/invariants/client/application/data/AccessLevels-Invariants.md [hunk 2]
```diff
@@ -1,14 +1,11 @@
 # AccessLevels Invariants
 
+##### Note: Not implemented, will be implemented later
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/application/data/AccessLevels-Invariants.md:3]: I think these are in ApplicationAppManagerData.t.i.sol

File: docs/userGuides/development/invariants/client/application/AppManager-Invariants.md
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/application/AppManager-Invariants.md:1]: The link in this file is broken

File: docs/userGuides/development/invariants/client/application/data/Accounts-Invariants.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Accounts Invariants
 
+#### Note: Not implemented, todo later
+
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/application/data/Accounts-Invariants.md:4]: These are in ApplicationAppManagerData.t.i.sol. Are these specific tests missing?

File: docs/userGuides/development/invariants/client/application/data/PauseRules-Invariants.md [hunk 4]
```diff
@@ -1,11 +1,9 @@
 # Pause Rules Invariants
 
+##### Note: Not implemented, todo
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/application/data/PauseRules-Invariants.md]: I think These are in ApplicationAppManagerData.t.i.sol

File: docs/userGuides/development/invariants/client/application/data/RiskScores-Invariants.md [hunk 5]
```diff
@@ -1,12 +1,13 @@
 # Risk Scores Invariants
 
+##### Note: Not implemented todo
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/application/data/RiskScores-Invariants.md:3]: I think these are in ApplicationAppManagerData.t.i.sol

File: docs/userGuides/development/invariants/client/application/data/Tags-Invariants.md [hunk 6]
```diff
@@ -1,14 +1,13 @@
 # Tags Invariants
 
+###### Note: Not implemented, todo
+
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/application/data/Tags-Invariants.md:4]: I think these are in ApplicationAppManagerData.t.i.sol

File: docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md [hunk 7]
```diff
@@ -1,5 +1,34 @@
...
+  
+
+## [Burnable ERC721 Invariants](../../../../../../../test/client/token/ERC721/invariant/ApplicationERC721MintBurn.t.i.sol)
+- The burn function should destroy tokens and reduce the total supply
+- A burned token should not be transferrable
+- approve() should revert if the token is burned
+- getApproved() should revert if the token is burned
+- ownerOf() should revert if the token has been burned.
+
+## [Mintable ERC721 Invariants]((../../../../../../../test/client/token/ERC721/invariant/ApplicationERC721MintBurn.t.i.sol))
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md]: This link is broken

File: docs/userGuides/development/invariants/protocol/Protocol-invariants.md [hunk 8]
```diff
@@ -66,7 +66,7 @@ All invariants have links to their invariant test files unless specified otherwi
 
 - When this rule is applied, an NFT can never be transferred before the hold-time period has passed counting from the last trade date.
 
-## Rule Invariants - Withour Stat Variables
+## Rule Invariants - Without Stat Variables
```
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/protocol/Protocol-invariants.md]: This should be "State Variables"

**Review by ShaneDuncan602 — CHANGES_REQUESTED** [review]
Thanks for the quick turnaround on this. Just found a few minor things. Great job, overall

File: docs/userGuides/development/invariants/client/application/data/AccessLevels-Invariants.md (hunk 2 above)
Comment (VoR0220) [reply docs/userGuides/development/invariants/client/application/data/AccessLevels-Invariants.md:3]: I don't think so. These invariants specifically are not implemented in that file. There is one test in invariants for removeAccessLevel and it's for `If removeAccessLevel is called by an account that is not an Access Level Admin the transaction will be reverted.` Likewise for the addAccessLevelToMultipleAccounts and addLevel.

File: docs/userGuides/development/invariants/client/application/AppManager-Invariants.md
Comment (VoR0220) [reply docs/userGuides/development/invariants/client/application/AppManager-Invariants.md:1]: Fixed

File: docs/userGuides/development/invariants/client/application/data/Accounts-Invariants.md (hunk 3 above)
Comment (VoR0220) [reply docs/userGuides/development/invariants/client/application/data/Accounts-Invariants.md:4]: Correct.

File: docs/userGuides/development/invariants/client/application/data/PauseRules-Invariants.md (hunk 4 above)
Comment (VoR0220) [reply docs/userGuides/development/invariants/client/application/data/PauseRules-Invariants.md]: The only pause rule invariants that have been implemented are the ones subject to the rule admin calls. Would you like me to say that these are sufficiently fuzz tested?

File: docs/userGuides/development/invariants/client/application/data/RiskScores-Invariants.md (hunk 5 above)
Comment (VoR0220) [reply docs/userGuides/development/invariants/client/application/data/RiskScores-Invariants.md:3]: Same as the above. addScore isn't really tested anywhere. addRiskScoreToMultipleAccounts only does an invariant for whether there's a proper admin adding the risk score. removeScore is not tested.

File: docs/userGuides/development/invariants/client/application/data/Tags-Invariants.md (hunk 6 above)
Comment (VoR0220) [reply docs/userGuides/development/invariants/client/application/data/Tags-Invariants.md:4]: Similar to the above but I would say we could get away with saying that these are sufficiently fuzz tested. Would you like me to do that?

File: docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md (hunk 7 above)
Comment (VoR0220) [reply docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md]: fixed

File: docs/userGuides/development/invariants/protocol/Protocol-invariants.md (hunk 8 above)
Comment (VoR0220) [reply docs/userGuides/development/invariants/protocol/Protocol-invariants.md]: fixed

**Review by ShaneDuncan602 — APPROVED** [review]

**Pull request merged**